### DISCUSSION TITLE
Strapi Beta Version and Future Upgrade Capabilities

### DISCUSSION THREAD

**Discussion by arnonzooz** [strapi/strapi#6330] [Legacy Topics]
Hi,

First, I'd like to say that I'm very much enjoying Strapi! I'm in the process of developing my first project, currently running 3.0.0-beta.20.3.

I have three questions:

1. Given that this is a Beta version, what are your suggestions in terms of using this version for deploying a project to production? Should I expect major changes between this beta version and the initial official release? Also, how easy will it be to upgrade? I'd very much appreciate some advice on this topic, since I am in a development phase now and would like to anticipate what's coming.

2. When installing Strapi, I see quite a number of dependency warnings, mostly related to new package versions being available. For example: `npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.`. Is this something you will be addressing in future versions? 

3. When is the official (non-beta) release due?

Appreciate your feedback.

Thanks

Arnon

**Comment by robertpiosik** [comment]
Ad 1. Upgrades are most often pretty easy. You just have to be careful to find and change all places where breaking changes are introduced. If you skip some releases you have to go through them sequentially and it can happen to be time consuming as the amount of custom code you write increases. I would not be so afraid though. These releases are now rather late than early. 

Migrations guides are located [here](https://strapi.io/documentation/3.0.0-beta.x/migration-guide/#migrating-from-alpha).

**Comment by derrickmehaffy** [comment]
I have been testing the rc.x releases the past 2 weeks (these are what will become the "stable" version) and yes there will be quite a bit of configuration changes as Strapi is moving onto the new configuration system. As for migrating/upgrading, I have been testing it out on my own project (to help the Strapi team validate and point out anything missed when they get around to making the migration guide).

All in all, it's not a difficult process, just a different structure than what you would be used to in the beta. If you have no plans to go into "production" in the next month, I'd say go ahead and start testing with the rc.x builds. Given the deployed documentation is not up to date with the rc.x releases, so you will need to clone the master branch and work with the local documentation. https://github.com/strapi/strapi/tree/master/docs

**Comment by arnonzooz** [comment]
OK, thanks. I think I'll go with your suggestion and start working with the rc.x builds. That will make it also easier to migrate my setup to the final version when it's released.

**Comment by jaarong** [comment]
New to node.js and strapi, but just wanted to ask if it were possible to simply upgrade strapi by deleting the folder and installing the new version then connecting it up with your current db? I am not customizing strapi's code right now, so it seems like that would be an option instead of the time-consuming migrations.

**Reply by derrickmehaffy** [reply]
Not currently no, Strapi cannot build models based on the structure of the database alone.
